prov/sockets/src/sock_wait.c

```diff
@@ -62,8 +62,8 @@ int sock_wait_get_obj(struct fid_wait *fid, void *arg)
 		break;
 
 	case FI_WAIT_MUTEX_COND:
-		mut_cond.mutex = &wait->wobj.mutex_cond.mutex;
-		mut_cond.cond = &wait->wobj.mutex_cond.cond;
+		mut_cond.mutex = (struct pthread_mutex_t*)&wait->wobj.mutex_cond.mutex;
+		mut_cond.cond = (struct pthread_cond_t*)&wait->wobj.mutex_cond.cond;
```

Comment:
in this case there is compilation warning, because ptherad types were declared 'forwarded'